sql_export/sql.py

```diff
@@ -0,0 +1,135 @@
+# -*- coding: utf-8 -*-
+##############################################################################
+#
+#    OpenERP, Open Source Management Solution
+#    Copyright (C) 2015 Akretion (<http://www.akretion.com>).
+#
+#    This program is free software: you can redistribute it and/or modify
+#    it under the terms of the GNU Affero General Public License as
+#    published by the Free Software Foundation, either version 3 of the
+#    License, or (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+
+from openerp.osv.orm import Model
+from openerp.osv import fields, orm
+from openerp.tools.translate import _
+import StringIO
+import base64
+import time
+
+
+PROHIBITED_WORDS = [
+    'delete',
+    'drop',
+    'insert',
+    'alter',
+    'truncate',
+    'execute',
+]
+
+
+class SqlExport(Model):
+    _name = "sql.export"
+    _description = "SQL export"
+
+    def _check_query_allowed(self, cr, uid, ids, context=None):
+        for obj in self.browse(cr, uid, ids, context=context):
+            if any(word in obj.query.lower() for word in PROHIBITED_WORDS):
+                return False
+            if obj.query.strip().split(' ')[0].lower() != 'select':
```

Comment:
Thanks for your comment!
You are right, checking first word is a bit restrictive.
But I am not sure checking for prohibited words is enought.
I could forget one or two important since I am not a huge expert in sql
Also, I did not include some word like update or create in this list.
So someone could update any tables, which we do not want!

I chose not to put create because of the field create_date that we may want to extract.
Also update, is easily used in fields name. (like in prestashop or magento connector)
That is why I restrict on the word SELECT.

Of course, I am open to idea to make it better.